R/ocjr-drug-court-SQ780.R

```diff
@@ -1,17 +1,522 @@
+    sq780_props = if_else(
+      # UCCS code or regex indicating 780 property crime, and no disqualifying regex
+      (uccs_property | regex_props) & !disqualifying_regex, TRUE, FALSE),
+    # Classify by specific statute ---------------------------------------------
+    # Simple possession
+    sq780_statute = case_when(
+      sq780_drugs ~
+        "63 O.S. Section 2-402 -- Simple Drug Possession",
+      # Grand / petit larceny
+      larceny & !receive_stolen ~
+        "21 O.S. Section 1704, 1705 -- Grand/Petit Larceny",
+      # Buying / receiving stolen property
+      larceny & receive_stolen ~
+        "21 O.S. Section 1713 -- Buying/Receiving/Concealing Stolen Property",
+      # Domesticated fish / game -- using new regex since it's split up above
+      str_detect(count_as_disposed, "(?i)fish|(?i)domesticated game") ~
+        "21 O.S. Section 1719.1 -- Taking Domesticated Gish/Game",
+      # Taking crude oil / gas / related
+      str_detect(count_as_disposed, "(?i)oil|drilling|gas") ~
+        "21 O.S. Section 1722 -- Unlawfully Taking Oil/Gas",
+      # Larceny of merchandise in retail or wholesale establishment
+      larceny & str_detect(count_as_disposed, "(?i)meat|edible|retail|wholesale") ~
+        "21 O.S. Section 1731 -- Larceny of Merchandise from Retail or Wholesale",
+      # Embezzlement
+      embezzle ~
+        "21 O.S. Sections 1451 -- Embezzlement",
+      # Defrauding hotel / lodging
+      hospitality ~
+        "21 O.S. Sections 1503 -- Defrauding Hospitality/Lodging",
+      # Vehicle Embezzlement
+      embezzle & str_detect(count_as_disposed, "(?i)vehicle|motor") ~
+        "21 O.S. Sections 1521 -- Vehicle Embezzlement",
+      # Fraud
+      str_detect(count_as_disposed, "cheat|defraud|personat") ~
+        "21 O.S. Sections 1541.1, 1541.2, 1541.3 -- Fraud",
+      # Pawn Shop Fraud
+      pawn ~
+        "59 O.S. Section 1512 -- Defrauding Pawn Shop",
+      # Forgery
+      forgery ~
+        "21 O.S. Section 1579, Section 1621 -- Forgery",
+      # Default
+      TRUE ~ "Unknown / Other"
+    )
+  ) |>
+  # I'm also gonna remove the filter for now so we can see what got classified as non-780
+  # filter(sq780_drugs | sq780_props) |>
+  select(id, case_type, date_filed, count_as_disposed, disposition_date,
+         disposition, charge_desc, offense_category_desc, offense_type_desc,
+         probability, uccs_code, party, uccs_drug_plus, poss, drug,
+         uccs_property, larceny, receive_stolen, embezzle, misc, pawn,
+         hospitality, forgery,
+         # keeping my new vars as well
+         sq780_drugs, sq780_props, sq780_statute) |>
+  # There are a few of these cluttering things up
+  filter(count_as_disposed != "")
+
+# Looking at the cleaned data a bit closer =====================================
+# Drugs first... ---------------------------------------------------------------
+clean_df |>
+  filter(sq780_drugs) |>
+  count(count_as_disposed, sort = T) |>
+  print(n = 200)
```

Comment:
My usual workflow for this regex type stuff is like 1) make best guess at regex -> 2) print the top 100-200 most common entries and scan for issues -> 3) fix issues in the regex, then go back to step 2 and repeat until you're satisfied with the results. So this is just for that.